packages/core/admin/admin/src/hooks/useMenu/index.js

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useReducer, useRef } from 'react';
+import { useRBACProvider, useAppInfos, useStrapiApp } from '@strapi/helper-plugin';
+import getPluginSectionLinks from './utils/getPluginSectionLinks';
+import getGeneralLinks from './utils/getGeneralLinks';
+import reducer, { initialState } from './reducer';
+
+const useMenu = () => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const { allPermissions } = useRBACProvider();
+  const { shouldUpdateStrapi } = useAppInfos();
+  const { menu } = useStrapiApp();
+
+  // We are using a ref because we don't want our effect to have this in its dependencies array
+  const generalSectionLinksRef = useRef(state.generalSectionLinks);
+  const shouldUpdateStrapiRef = useRef(shouldUpdateStrapi);
+  // Once in the app lifecycle the menu should not be added into any dependencies array
+  const menuRef = useRef(menu);
```

Comment:
I'm not sure why this is necessary. From what I see, it only registers the ref on the first render and is never reassigned later. Also, the refs are not returned by the hook.

Plus, they are mostly used in `resolvePermissions` which is re-created on every render.s


Maybe I'm missing smth 🤔

Reply:
No you are right it's useless!